### PR TITLE
Sing up

### PR DESCRIPTION
**Pull request by francomino-wolox**
## Summary
Add sign up.
feat: validator for fields user when create a new user.
feat: log errores.
add: folder helper and functions.

## Known Issues

nothing

## Trello Card

https://trello.com/c/YhcJ1llQ

### CODE REVIEW

File: app/helpers/crypto.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import bcryptjs from 'bcryptjs';
+
+export function encrypt(data: string): string {
+  const salt: string = bcryptjs.genSaltSync(10);
+  return bcryptjs.hashSync(data, salt);
```
Comment (DiegoWolox) [inline app/helpers/crypto.ts:5]: please, use the async version

File: app/helpers/crypto.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+export function compareEncrypt(hash: string, data: string): boolean {
+  if (hash && data) {
+    return bcryptjs.compareSync(data, hash);
```
Comment (DiegoWolox) [inline app/helpers/crypto.ts:10]: same here

**Review by DiegoWolox — APPROVED** [review]

File: app/helpers/crypto.ts (hunk 1 above)
Comment (francomino-wolox) [reply app/helpers/crypto.ts:5]: changed in next card

File: app/helpers/crypto.ts (hunk 2 above)
Comment (francomino-wolox) [reply app/helpers/crypto.ts:10]: changed in next card

**Pull request merged**